content/guides/getstarted/scaffold-nextjs-anchor.md

```diff
@@ -0,0 +1,306 @@
+│
+◆  Enter project name
+│  - hello_solana
+└
+```
+
+### Web preset
+
+Then you will be asked for a preset to use, could be Next.JS or React + React
+Router DOM
+
+I prefer nextJS as it has file based routing and app routing. You can go ahead
+with React + React Router DOM if you prefer that.
+
+```
+◆  Select a preset
+│  ● Next.js
+│  ○ React + React Router DOM
+```
+
+### UI library
+
+After you pick a framework, you'll be prompted to select a styling library.
+
+```
+◆  Select a UI library
+│  ○ None
+│  ● Tailwind
+└
+```
+
+We use tailwindCSS and Daisy UI here, but you can choose none if you already
+have your preferred one.
+
+### Anchor Template
+
+Next this scaffold CLI asks which template for anchor you want to use. Anchor is
+a sealevel framework for writing solana programs.
+
+```
+◆  Select an Anchor template
+│  ○ counter
+│  ● hello-world
+│  ○ none
+└
+```
+
+After waiting for some time, you can make some popcorn so that you can eat those
+when the scaffolding is ready (Installing takes ~3 mins).
+
+```
+◓  Creating new workspace with npm...
+```
+
+When you are done you can change the terminal directory to your project.
+
+```
+◇  Successfully installed preset @solana-developers/preset-next@2.0.1.
+│
+└  Run `cd ./hello_solana-app` to get started.
+```
+
+type `pwd` to check if your path is correct, then type `code .` to open VSCode
+inside the scaffolded project.
+
+In your scaffold, five commands are waiting for you to execute whenever you
+like. Type `npm run` to see all the commands available.
+
+```
+Scripts available in @counter-app/source@0.0.0 via `npm run-script`:
+  anchor
+    nx run anchor:anchor
+  anchor-build
+    nx run anchor:anchor build
+  anchor-localnet
+    nx run anchor:anchor localnet
+  anchor-test
+    nx run anchor:anchor test
+  build
+    nx build web
+  dev
+    nx serve web
+```
+
+To run any of these commands, type `npm run <SCRIPT_COMMAND>` , for example,
+`npm run dev` to start the development server of your website or
+`npm run anchor-test` to run tests on your scaffolded solana program.
+
+## 3. Experience hands-on with scaffolds
+
+### Anchor Program
+
+Before starting, please initiate your project with git init to track the changes
+you make. Here is what our directory tree looks like at the root of our project.
+
+```
+    hello_solana
+    |
+    ├── anchor
+    │   ├── migrations
+    │   ├── programs
+		|   |   └── hello_world
+    │   └── tests
+    ├── web
+    │   ├── app
+    │   ├── components
+        └── public
+```
+
+Inside `anchor > programs > hello-world > src > lib.rs` , we can start writing
+our instructions.
+
+To build this program execute `npm run anchor-build`; your IDL generated in the
+target folder. IDL is a spec similar to ABI in EVM environments. It describes
+how the JSON file will interact with a solana program. copy the address inside
+metadata from your `hello_world.json` to your anchor program
+`[lib.rs](http://lib.rs)` line 3 and paste your address there.
```

Comment:
```suggestion
To build this program execute `npm run anchor-build`. Your Anchor project will be built, and an IDL generated in the target folder. IDL is a spec that describes the instruction handler in your program and their arguments. Copy the address inside
metadata from your `hello_world.json` to the `program_id` section in your Anchor program's `lib.rs`.
```

Reply:
> IDL is a spec similar to ABI in EVM environments. It describes
how the JSON file will interact with a solana program.

Please don't put EVM-requiring explanations before explanations that don't require EVM knowledge! Most people learning Solana will not know what EVM is.

Reply:
Re:

> Copy the address inside
metadata from your `hello_world.json` to the `program_id` section in your Anchor program's `lib.rs`.

(I made this say `program_id`, before it just said 'line 3'.

Is this actually necessary? Why would the IDL have a different program ID?